### PR TITLE
added spatial query and fixed some issues

### PR DESCRIPTION
**Pull request by yantrabuddhi**
added spatial query to timeoctomap

### CODE REVIEW

**Comment by ngeiswei** [comment]
@yantrabuddhi I suppose you want me to review merge that, right?

File: opencog/timeoctomap/pointmemory/PointMemorySCM.cc [hunk 1]
```diff
@@ -148,10 +155,8 @@ bool PointMemorySCM::map_ato(const string& map_name,Handle ato,double x,double y
 
 Handle PointMemorySCM::get_first_ato(const string& map_name,Handle ato,int elapse)
 {
-    time_pt tpt,tp; duration_c dr;
-    tsa[map_name]->get_current_time_range(tpt,dr);
-    dr=std::chrono::milliseconds(elapse);
-    tpt-=dr;
+    time_pt tpt,tp; 
+    if (!get_map_time(map_name,elapse,tpt))return UndefinedHandle;
```
Comment (ngeiswei) [inline opencog/timeoctomap/pointmemory/PointMemorySCM.cc]: better put a space or a newline before return.

**Comment by ngeiswei** [comment]
I didn't get deep into it, but on the surface it looks good, just put a space before return here and there.

**Comment by yantrabuddhi** [comment]
Hi Nil,
 I formatted the code as you said. I put a newline in each case.
Regards,
Mandeep

**Pull request merged**